src/odemis/dataio/csv.py

```diff
@@ -132,6 +132,26 @@ def export(filename, data):
             csv_writer.writerow(headers)
             csv_writer.writerows(rows)
 
+    elif data.metadata.get(model.MD_ACQ_TYPE, None) == model.MD_AT_EK:
+        logging.debug("Exporting angular spectrum data to CSV")
+
+        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
+        if unit_c == "m":
+            spectrum_range = [s * 1e9 for s in spectrum_range]
+            unit_c = "nm"
+
+        angle_range, _ = spectrum.get_angle_range(data)
+        unit_a = "°"
+        angles = [math.degrees(theta) for theta in angle_range if not math.isnan(theta)]  # Convert radians to degrees
+
```

Comment:
Please add a `assert len(angles) == len(data)`.
Because of the `zip()`, if they are not the same length, the shortest length will be used and the rest of the data discarded. That should never happen!